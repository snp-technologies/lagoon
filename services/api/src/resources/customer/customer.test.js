const Sql = require('./sql');

describe('Sql', () => {
  describe('updateCustomer', () => {
    it('if not admin, should create statement with IN clause', () => {
      const cred = {
        permissions: {
          customers: [1, 2, 3],
        },
      };
      const input = {
        id: 1,
        patch: {
          name: 'test',
        },
      };

      const ret = Sql.updateCustomer(cred, input);
      expect(ret).toMatchSnapshot();
    });

    it('if admin, should not create IN clause', () => {
      const cred = {
        role: 'admin',
        permissions: {},
      };
      const input = {
        id: 1,
        patch: {
          name: 'test',
        },
      };

      const ret = Sql.updateCustomer(cred, input);
      expect(ret).toMatchSnapshot();
    });
  });
  describe('selectCustomer', () => {
    it('should create a proper query', () => {
      const ret = Sql.selectCustomer(1);

      expect(ret).toMatchSnapshot();
    });
  });
  describe('selectCustomerByName', () => {
    it('if not admin, should create statement with IN clause', () => {
      const cred = {
        role: 'user',
        permissions: {
          customers: [1, 2],
        },
      };

      const ret = Sql.selectCustomerByName(cred, 'c1');
      expect(ret).toMatchSnapshot();
    });

    it('if admin, should not create IN clause', () => {
      const cred = {
        role: 'admin',
        permissions: {},
      };

      const ret = Sql.selectCustomerByName(cred, 'c1');
      expect(ret).toMatchSnapshot();
    });
  });
});
