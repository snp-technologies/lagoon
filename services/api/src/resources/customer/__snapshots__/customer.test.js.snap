// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql selectCustomerByName if admin, should not create IN clause 1`] = `"select * from \`customer\` where \`name\` = 'c1'"`;

exports[`Sql selectCustomerByName if not admin, should create statement with IN clause 1`] = `"select * from \`customer\` where \`name\` = 'c1' and \`id\` in (1, 2)"`;

exports[`Sql selectCustomer should create a proper query 1`] = `"select * from \`customer\` where \`id\` = 1"`;

exports[`Sql updateCustomer if admin, should not create IN clause 1`] = `"update \`customer\` set \`name\` = 'test' where \`id\` = 1"`;

exports[`Sql updateCustomer if not admin, should create statement with IN clause 1`] = `"update \`customer\` set \`name\` = 'test' where \`id\` = 1"`;
